notebooks/04_model_select_and_optimize/azureml_hyperdrive_tuning_svd.ipynb

```diff
@@ -0,0 +1,731 @@
+    "    max_concurrent_runs=MAX_CONCURRENT_RUNS\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "0dc48cf352a74c58a2dc976ff33421fa",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "_HyperDriveWidget(widget_settings={'childWidgetDisplay': 'popup', 'send_telemetry': False, 'log_level': 'INFO'…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "RunId: movielens_100k_svd_model_1550672871383\n",
+      "\n",
+      "Execution Summary\n",
+      "=================\n",
+      "RunId: movielens_100k_svd_model_1550672871383\n",
+      "\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "{'runId': 'movielens_100k_svd_model_1550672871383',\n",
+       " 'target': 'cpuclustersvd',\n",
+       " 'status': 'Completed',\n",
+       " 'endTimeUtc': '2019-02-20T15:00:23.000Z',\n",
+       " 'properties': {'primary_metric_config': '{\"name\": \"precision_at_k\", \"goal\": \"maximize\"}',\n",
+       "  'runTemplate': 'HyperDrive',\n",
+       "  'azureml.runsource': 'hyperdrive'},\n",
+       " 'logFiles': {'azureml-logs/hyperdrive.txt': 'https://anargyri6699545766.blob.core.windows.net/azureml/ExperimentRun/dcid.movielens_100k_svd_model_1550672871383/azureml-logs/hyperdrive.txt?sv=2018-03-28&sr=b&sig=dt%2BtkHqz49%2F2sQTmTzHDMh%2BchdRdiMSELDnQycGsE3g%3D&st=2019-02-20T14%3A50%3A26Z&se=2019-02-20T23%3A00%3A26Z&sp=r'}}"
+      ]
+     },
+     "execution_count": 25,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Create an experiment to track the runs in the workspace\n",
+    "exp = aml.core.Experiment(workspace=ws, name=EXP_NAME)\n",
+    "run = exp.submit(config=hd_config)\n",
+    "\n",
+    "azureml.widgets.RunDetails(run).show()\n",
+    "run.wait_for_completion(show_output=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Get best run and printout metrics\n",
+    "best_run = run.get_best_run_by_primary_metric()\n",
+    "\n",
+    "best_run_metrics = best_run.get_metrics()\n",
+    "parameter_values = best_run.get_details()['runDefinition']['Arguments']"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 27,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'Number of epochs': 30,\n",
+       " 'rmse': 1.1324249983589592,\n",
+       " 'precision_at_k': 0.17932131495228,\n",
+       " 'ndcg_at_k': 0.20470147010861087}"
+      ]
+     },
+     "execution_count": 27,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "best_run_metrics"
```

Comment:
Here the metrics are much worse than the ones in the original notebook https://github.com/Microsoft/Recommenders/blob/master/notebooks/02_model/surprise_svd_deep_dive.ipynb. Do you know why?

Reply:
No, precision is much better. This is what we care about because the tuning maximizes precision. It is not surprising that the other metrics are worse, since they may not be strongly correlated with precision.

Reply:
I can elaborate more in the concluding remarks.

Reply:
that makes sense